task2/InterfaceManager.py

```diff
@@ -0,0 +1,109 @@
+import json
+import logging
+from dataclasses import dataclass, field
+from typing import List
+
+import marshmallow as mm
+import requests
+from dataclasses_json import DataClassJsonMixin, config
+from marshmallow import fields, EXCLUDE
+
+# Logger
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+formatter = logging.Formatter("%(asctime)s - %(levelname)s - %(message)s")
+
+file_handler = logging.FileHandler('output_folder/interfaces.log', mode='w')
+file_handler.setLevel(logging.INFO)
+file_handler.setFormatter(formatter)
+
+logger.addHandler(file_handler)
+
+
+@dataclass
+class AddressIpv4Inside(DataClassJsonMixin):
+    ip: str
+    netmask: str
+
+
+@dataclass
+class AddressIpv6Inside(DataClassJsonMixin):
+    ip: str
+    prefix_length: int = field(
+        metadata=config(mm_field=mm.fields.Number(data_key='prefix-length')))
+
+
+@dataclass
+class AddressIpv4(DataClassJsonMixin):
+    address: List[AddressIpv4Inside] = field(
+        metadata=config(mm_field=mm.fields.List(fields.Nested(AddressIpv4Inside.schema(unknown=mm.EXCLUDE)))),
+        default_factory=list)
+
+
+@dataclass
+class AddressIpv6(DataClassJsonMixin):
+    address: List[AddressIpv6Inside] = field(
+        metadata=config(mm_field=mm.fields.List(fields.Nested(AddressIpv6Inside.schema(unknown=mm.EXCLUDE)))),
+        default_factory=list)
```

Comment:
@patrikprovaznik these are extra unneeded classes.. you can do list of AddressIpv6Inside directly in Interface dataclass..